sklearn/metrics/classification.py

```diff
@@ -1091,6 +1234,41 @@ def _prf_divide(numerator, denominator, metric, modifier, average, warn_for):
     return result
 
 
+_ALL_ZERO = object()  # sentinel for special, degenerate case
+
+
+def _check_set_wise_labels(y_true, y_pred, average, labels, pos_label):
```

Comment:
Do you have time to look at #13143 ? Currently, we fail to raise consistent warning about the edge cases of P/R/F.

Reply:
You're right. We shouldn't be specially handling the all-zeros case here.